pkg/analysis/markerscope/initializer_test.go

```diff
@@ -0,0 +1,350 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package markerscope_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"sigs.k8s.io/kube-api-linter/pkg/analysis/initializer"
+	"sigs.k8s.io/kube-api-linter/pkg/analysis/markerscope"
+)
+
+var _ = Describe("markerscope initializer", func() {
+	Context("config validation", func() {
+		type testCase struct {
+			config      markerscope.MarkerScopeConfig
+			expectedErr string
+		}
+
+		DescribeTable("should validate the provided config",
+			func(in testCase) {
+				ci, ok := markerscope.Initializer().(initializer.ConfigurableAnalyzerInitializer)
+				Expect(ok).To(BeTrue())
+
+				errs := ci.ValidateConfig(&in.config, field.NewPath("markerscope"))
+				if len(in.expectedErr) > 0 {
+					Expect(errs.ToAggregate()).To(HaveOccurred())
+					Expect(errs.ToAggregate().Error()).To(ContainSubstring(in.expectedErr))
+				} else {
+					Expect(errs).To(HaveLen(0), "No errors were expected")
+				}
+			},
+
+			Entry("With nil config", testCase{
+				config:      markerscope.MarkerScopeConfig{},
+				expectedErr: "",
+			}),
+
+			Entry("With empty config", testCase{
+				config:      markerscope.MarkerScopeConfig{},
+				expectedErr: "",
+			}),
+
+			Entry("With valid warn policy", testCase{
+				config: markerscope.MarkerScopeConfig{
+					Policy: markerscope.MarkerScopePolicyWarn,
+				},
+				expectedErr: "",
+			}),
+
+			Entry("With valid suggest_fix policy", testCase{
+				config: markerscope.MarkerScopeConfig{
+					Policy: markerscope.MarkerScopePolicySuggestFix,
+				},
+				expectedErr: "",
+			}),
+
+			Entry("With invalid policy", testCase{
+				config: markerscope.MarkerScopeConfig{
+					Policy: "invalid-policy",
+				},
+				expectedErr: `markerscope.policy: Invalid value: "invalid-policy": invalid policy, must be one of: "Warn", "SuggestFix"`,
+			}),
+
+			Entry("With valid marker rules", testCase{
+				config: markerscope.MarkerScopeConfig{
+					CustomMarkers: []markerscope.MarkerScopeRule{
+						{
+							Identifier: "custom:marker",
+							Scopes:     []markerscope.ScopeConstraint{markerscope.FieldScope},
+						},
+					},
+				},
+				expectedErr: "",
+			}),
+
+			Entry("With marker rule having empty scope", testCase{
+				config: markerscope.MarkerScopeConfig{
+					CustomMarkers: []markerscope.MarkerScopeRule{
+						{
+							Identifier: "custom:marker",
+							Scopes:     []markerscope.ScopeConstraint{},
+						},
+					},
+				},
+				expectedErr: `scope is required`,
```

Comment:
Can you demonstrate the full error including the field path? Or do we not get that here?